File: frontend/src/types/list.ts
export interface ListResponse {
  id: string;
  name: string;
  description: string | null;
  schema_id: string | null;
  default_schema_id: string | null; // Workspace-wide schema (fields for all videos)
  video_count: number;
  created_at: string;
  updated_at: string;
}

export interface ListCreate {
  name: string;
  description?: string;
  schema_id?: string;
}

export interface ListUpdate {
  name?: string;
  description?: string;
  default_schema_id?: string | null;
}

File: backend/app/schemas/list.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ListCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: str | None = Field(None, max_length=1000)
    schema_id: UUID | None = None
    user_id: UUID | None = None  # Will be set from authenticated user


class ListUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    description: str | None = None
    default_schema_id: UUID | None = None  # Workspace-wide schema for all videos


class ListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    description: str | None
    user_id: UUID
    schema_id: UUID | None
    default_schema_id: UUID | None = None  # Workspace-wide schema for all videos
    video_count: int
    created_at: datetime
    updated_at: datetime
